### PR TITLE
refactor: supercharge Bulk actions

### PR DESCRIPTION
**Pull request by ruthra-kumar**
Bulk Processing feature was poorly designed, due to which its performance deteriorates as more and more actions are taken in a single day. With the redesign, we've improved the normal processing speed significantly and removed the performance degradation completely.

## Details:

1. `Bulk Transaction Log` has been converted to a virtual doctype and has simple success and fail count, and UI button to retry for failed documents.
2. `Bulk Transaction Log Detail` has been converted from child to a normal doctype.
3. `update_logger` method, which updated the entire `Bulk Transaction Log` document for every document, which directly caused the poor performance, has been removed completely. It has been replaced with `create_log` which is just a simple method to create `Bulk Transaction Log Detail`

## Performance
Compared to the `26.668 s` from the [previous refactor](https://github.com/frappe/erpnext/pull/37971), this `6.74 s` is a 74% reduction in runtime for processing the same volume of documents.
```
In [1]: sinvs=frappe.db.get_all("Sales Invoice", filters={"status":"Overdue"}, limit=100)

In [2]: from erpnext.utilities.bulk_transaction import job

In [3]: %time job(sinvs, "Sales Invoice", "Payment Entry")
CPU times: user 3.44 s, sys: 200 ms, total: 3.64 s
Wall time: 6.74 s
```

## Demo

https://github.com/frappe/erpnext/assets/3272205/c93f56af-688e-41f3-9781-ef0acfb9fea9

Internal Ref: [5034](https://support.frappe.io/helpdesk/tickets/5034), [5626](https://support.frappe.io/app/hd-ticket/5626)

### CODE REVIEW

File: erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.js [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
+		}, __("View"));
+		frm.add_custom_button(__('Failed Entries'), function() {
+			frappe.set_route('List', 'Bulk Transaction Log Detail', {'date': frm.doc.date, 'transaction_status': "Failed"});
+		}, __("View"));
+		if (frm.doc.failed) {
+			frm.add_custom_button(__('Retry Failed Transactions'), function() {
+				frappe.call({
+					method: "erpnext.utilities.bulk_transaction.retry",
+					args: {date: frm.doc.date}
+				}).then(()=> {	});
```
Comment (deepeshgarg007) [inline erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.js]: `.then` is not needed if we are not doing anything within it.

File: erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py [hunk 2]
```diff
@@ -1,67 +1,105 @@
...
-				status="Failed",
-				log_date=log_date,
-				restarted=1,
-			)
+	def db_insert(self, *args, **kwargs):
+		pass
+
+	def load_from_db(self):
+		log_detail = qb.DocType("Bulk Transaction Log Detail")
+		succeeded_logs = (
```
Comment (deepeshgarg007) [inline erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py:25]: This type of query might give an index error if there are no `Bulk Transaction Log Detail` and the result is `None`.

File: erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.js (hunk 1 above)
Comment (ruthra-kumar) [reply erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.js]: removed

File: erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py (hunk 2 above)
Comment (ruthra-kumar) [reply erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py:25]: `DoesNotExistError` exception will be raised, if an invalid date or if there are no logs on that date.
Screenshot 2023-11-13 at 10 43 20 AM

**Pull request merged**